### ISSUE TITLE
1.22.1: build fails on linkig 

### ISSUE THREAD

**Issue by kloczek** [c-ares/c-ares#629]
Looks like build fails on linking arestest
```console
[ 71%] Linking CXX executable ../bin/arestest
cd /home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/x86_64-redhat-linux-gnu/test && /usr/bin/cmake -E cmake_link_script CMakeFiles/arestest.dir/link.txt --verbose=1
/usr/bin/g++ -Wl,--gc-sections -Wl,--as-needed -flto=auto -flto-partition=none -fuse-linker-plugin -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,--build-id=sha1 "CMakeFiles/arestest.dir/ares-test-main.cc.o" "CMakeFiles/arestest.dir/ares-test-init.cc.o" "CMakeFiles/arestest.dir/ares-test.cc.o" "CMakeFiles/arestest.dir/ares-test-ns.cc.o" "CMakeFiles/arestest.dir/ares-test-parse.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-a.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-aaaa.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-caa.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-mx.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-naptr.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-ns.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-ptr.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-soa.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-soa-any.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-srv.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-txt.cc.o" "CMakeFiles/arestest.dir/ares-test-parse-uri.cc.o" "CMakeFiles/arestest.dir/ares-test-misc.cc.o" "CMakeFiles/arestest.dir/ares-test-live.cc.o" "CMakeFiles/arestest.dir/ares-test-mock.cc.o" "CMakeFiles/arestest.dir/ares-test-mock-ai.cc.o" "CMakeFiles/arestest.dir/ares-test-internal.cc.o" "CMakeFiles/arestest.dir/dns-proto.cc.o" "CMakeFiles/arestest.dir/dns-proto-test.cc.o" -o ../bin/arestest  -Wl,-rpath,/home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/x86_64-redhat-linux-gnu/lib64 ../lib64/libgmock.a ../lib64/libcares.so.2.8.1
/usr/bin/ld: CMakeFiles/arestest.dir/ares-test.cc.o: in function `ares::test::TempNam[abi:cxx11](char const*, char const*)':
ares-test.cc:(.text+0x5395): warning: the use of `tempnam' is dangerous, better use `mkstemp'
/usr/bin/ld: CMakeFiles/arestest.dir/ares-test-main.cc.o: relocation R_X86_64_32 against `.rodata' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/ld: failed to set dynamic section sizes: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [test/CMakeFiles/arestest.dir/build.make:470: bin/arestest] Error 1
```

**Comment by kloczek** [comment]
On first look it looks like on compile of ares-test-main.cc are completly is used global CFLAGS
```console
[tkloczko@pers-jacek x86_64-redhat-linux-gnu]$ rm ./test/CMakeFiles/arestest.dir/ares-test-main.cc.o; make
[..]

[ 71%] Building CXX object test/CMakeFiles/arestest.dir/ares-test-main.cc.o
cd /home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/x86_64-redhat-linux-gnu/test && /usr/bin/g++ -DHAVE_CONFIG_H=1 -DHAVE_USER_NAMESPACE -DHAVE_UTS_NAMESPACE -D_GNU_SOURCE -D_POSIX_C_SOURCE=199309L -D_XOPEN_SOURCE=600 -I/home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/x86_64-redhat-linux-gnu/test -I/home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/x86_64-redhat-linux-gnu -I/home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1 -I/home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/src/lib -I/home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/include -I/home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/test/include -I/home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/test/SYSTEM -I/home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/test/gmock-1.11.0 -std=c++11 -MD -MT test/CMakeFiles/arestest.dir/ares-test-main.cc.o -MF CMakeFiles/arestest.dir/ares-test-main.cc.o.d -o CMakeFiles/arestest.dir/ares-test-main.cc.o -c /home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/test/ares-test-main.cc
```
Here is set my set of rnv variables passed in cmake env
```console
[tkloczko@pers-jacek SPECS]$ rpm -E %set_build_flags

ASMFLAGS="-O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none";
CFLAGS="-O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none";
CXXFLAGS="-O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none";
FFLAGS="-O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -I/usr/lib64/gfortran/modules";
FCFLAGS="-O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -I/usr/lib64/gfortran/modules";
LDFLAGS="-Wl,--gc-sections -Wl,--as-needed -flto=auto -flto-partition=none -fuse-linker-plugin -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,--build-id=sha1";
RUSTFLAGS="-C codegen-units=1 -C debuginfo=2 -C opt-level=2 -C link-arg=-fdata-sections -C link-arg=-ffunction-sections -C link-arg=-Wl,--as-needed -C link-arg=-Wl,-z,now -C link-arg=-Wl,-z,relro --cap-lints=warn" ;
VALAFLAGS="-g" ;
CC="/usr/bin/gcc"; CXX="/usr/bin/g++"; FC="/usr/bin/gfortran";
AR="/usr/bin/gcc-ar"; NM="/usr/bin/gcc-nm"; RANLIB="/usr/bin/gcc-ranlib";
export ASMFLAGS CFLAGS CXXFLAGS FFLAGS FCFLAGS LDFLAGS VALAFLAGS CC CXX FC AR NM RANLIB RUSTFLAGS VALAFLAGS;
```
In compile options there are no traces of use CFLAGS
```console
CFLAGS="-O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none";
```
There is no LTO options and there is no `-specs=/usr/lib/rpm/redhat/redhat-hardened-cc1` which forces gcc to use by default `-fPIE`.

**Comment by kloczek** [comment]
Why ares-test-main.cc is compiled using custom Makelile files instead cmake files? 🤔 
```console
[tkloczko@pers-jacek c-ares-1.22.1]$ grep -r ares-test-main.cc
test/Makefile.in:TESTSOURCES = ares-test-main.cc        \
test/Makefile.inc:TESTSOURCES = ares-test-main.cc       \

**Comment by kloczek** [comment]
BTW why in test/gmock-1.11.0 is  bundled copy of old gtest? 🤔

**Comment by bradh352** [comment]
Is this a regression from a prior version?  What version last worked if so?

What distro/version are you running on?

> Why ares-test-main.cc is compiled using custom Makelile files instead cmake files? 🤔

Its not. c-ares supports multiple build systems, if you're using cmake, it won't use the Makefile.in, it does however extract the variable from Makefile.inc to know what files to build.

> BTW why in test/gmock-1.11.0 is bundled copy of old gtest? 🤔

Because the original author of the tests bundled it, and gtest removed the combined source file method for packaging gtest so no one has had the time to update the build systems to look for an external gtest.  We welcome contributions

**Comment by bradh352** [comment]
Also, `CFLAGS` wouldn't be used for building the test cases since they are c++. Can you tell if the `CFLAGS` are used when building the library itself which is all C?

I can, however, confirm that if I pass `CXXFLAGS` as an environment variable, its not honored on my system.  However, if you do it the "CMake" way, by passing the flags as `-DCMAKE_CXX_FLAGS="-blah"` on the cmake command line, it works as intended.  There's nothing we're doing in the cmake files to mess with CXX flags like that, so I'm not sure what to suggest.

**Comment by kloczek** [comment]
> Is this a regression from a prior version? What version last worked if so?

It was working in c-ares-1.20.1.

>Also, CFLAGS wouldn't be used for building the test cases since they are c++. Can you tell if the CFLAGS are used when building the library itself which is all C?

As you see I'm passing as well $CXXFLAGS.

> I can, however, confirm that if I pass CXXFLAGS as an environment variable, its not honored on my system.

OK so we have cause.

> However, if you do it the "CMake" way, by passing the flags as -DCMAKE_CXX_FLAGS="-blah" 

This is not standard way.
None of the other know me projects ignores env variable and uses what is passed over this option.

**Comment by bradh352** [comment]
Its odd, but I think https://github.com/c-ares/c-ares/commit/33ee6c0 fixes it.

**Comment by kloczek** [comment]
Cannot test this commit on top od dist tar ball because it is not possible to apply it on top of that tar ball.
To test it would be necessary to use tar ball generated from git tag and ..  here is small obstacle in tagging convention.
It would be good IMO to change tagging convention from current `cares-<version_with_underscores>` to `<version_with_dots>`.

Use tags aa `<version_with_dots>` would generate tar ball generated from git tag like last one https://github.com/c-ares/c-ares/archive/refs/tags/cares-1_22_1.tar.gz with base directory instead `cares-cares-1_22_1` just `cares-1.22.1` because base directory is automatically formed out os repo name and git tag.
Doable on next release? 🤔

**Comment by bradh352** [comment]
@bagder see above

**Comment by bagder** [comment]
The tarball GitHub generates is not the release package. It is just the source packaged from the same time stamp the release was made. The release tarballs are *generated*, signed and provided on c-ares.org.

The tag naming used in the git repository has been used since 2004 and I have stuck to it for consistency. Changing it now will make it more difficult to do automation involving tag/releases names through history.

**Comment by kloczek** [comment]
> The tarball GitHub generates is not the release package. It is just the source packaged from the same time stamp the release was made. The release tarballs are generated, signed and provided on c-ares.org.

Exactly what it is.
Only use tagged tar balls guarantees that on top of it will be possible to apply any sequence of patches added after release.

BTW why tagged tar ball cannot be dist tarball?
```console
[tkloczko@pers-jacek SPECS]$ grep "Source:              %{VCS}/archive/" *spec | wc -l; ls -1 *spec|wc -l
1826
4744
```
Above says that I have +4.7k rpm spec files which almost 40% uses directly tar balls
Adytony to make github allow use uch tar balls as verifiable.
https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-tags
That fictionality is more and more used because it takes many maintainers release process easier (just tag and sign and another steps).

> The tarball GitHub generates is not the release package. It is just the source packaged from the same time stamp the release was made. The release tarballs are generated, signed and provided on c-ares.org.
> The tag naming used in the git repository has been used since 2004 and I have stuck to it for consistency. Changing it now will make it more difficult to do automation involving tag/releases names through history.

I've not been asking since when is used current tagging convention.
My question was/; is it possible to use git tag convention like `<version_with_dots>` or `v<version_with_dots>` ? 🤔 
(like it is in case +95% of the git repos .. not only on github)
```console
[tkloczko@pers-jacek SPECS]$ grep "Source:              %{VCS}/archive/%{version}/%{name}-%{version}.tar.gz" *spec | wc -l
770
[tkloczko@pers-jacek SPECS]$ grep "Source:              %{VCS}/archive/v%{version}/%{name}-%{version}.tar.gz" *spec | wc -l
820
```
All to make people like me (packagers) a bit easier.
Please ..

**Comment by bagder** [comment]
>  Only use tagged tar balls guarantees that on top of it will be possible to apply any sequence of patches added after release.

No it doesn't. There is no guarantee that a future patch will apply against version x.y.z no matter how you fetched that version. The only way to guarantee that a patch applies is you first checkout the git commit immediately before a particular commit was made. In all other cases, you have to try and see.

> BTW why tagged tar ball cannot be dist tarball?

Because, as I already explained, the tarball is *generated*. It contains files (most obviously `configure`) that are not present in the git repository. They have been done like this since the first c-ares release.

It also makes the releases conveniently and nicely detached and independent of the git hosting service in use. The releases are made and provided exactly the same as before c-ares even used git.

> My question was/; is it possible to use git tag convention

I already answered why we use the current approach: consistency and sticking to what has been used for almost twenty years. Can it be changed? Of course it can. The question is not if it *can* be changed. The question is if we should change it.

Since we don't provide the actual release this way, it seems like a weak reason for changing the tag naming convention.

**Comment by kloczek** [comment]
> No it doesn't. There is no guarantee that a future patch will apply against version x.y.z no matter how you fetched that version. 

Can you show ONE example of that?
```console
[tkloczko@pers-jacek SPECS]$ grep "Patch.*%{VCS}" *spec | wc -l
5782
```
Above means that I have now integrated +5..7k patches generated directly from exact git commits hashes.
```console
[tkloczko@pers-jacek SPECS]$ grep "Patch.*%{VCS}" *spec -l | xargs grep -c "Patch.*%{VCS}" | sort -rn -k2 -t: | head -n 10
opensc.spec:366
libimobiledevice.spec:190
libiscsi.spec:187
qt5-qtbase.spec:143
groff.spec:129
mdadm.spec:125
dracut.spec:109
audit.spec:108
libgda.spec:73
ibus.spec:72
```
As you see in some most extreme cases I single build procedure I have almost 400 patches generated from git commits.
I'm using such integration +4 years and since then I was not able to find even ONE case which you've described.

> The only way to guarantee that a patch applies is you first checkout the git commit immediately before a particular commit was made. In all other cases, you have to try and see.

Things like that are not objective truth.
It may be true only because your personal decision about not storing exact source files in git repo.
Even dist tar ball which uses GNU auto tools adds generated files and should not change any files which are stored in VCS.
So question is: why in case of c-ares which seems on whole landscape of OSS projects kind of exceptions not all files in gist tar ball are the same with those in VCS? 🤔 

> I already answered why we use the current approach: consistency and sticking to what has been used for almost twenty years.

Things changed in last 20 yars.
Git gas been developed. People started in mean time integrating direcrly git repositories interractions with packaging processes is more and more used.

**Comment by bradh352** [comment]
well, the source code in the github package is going to be identical to the official release package.  The downfall of course is the github one isn't signed and doesn't have the necessary generated files for people using autotools or static makefiles for platforms that don't support autotools or cmake.

I don't think this can be securely automated really, hosting signing keys in github secrets for a github action to run just sounds like a compromise waiting to happen.

A distro using the unsigned github-generated tarball does sound suspicious to me from a security standpoint, so I'm not following along on that discussion point.

I also don't think I'm following the talk about importing specific future commits as patches on top of a release.  There's no reason why that wouldn't work against either the official or github tarballs.  Other than the normal issues with applying patches, such as if one patch/commit depends on another.

**Comment by kloczek** [comment]
> well, the source code in the github package is going to be identical to the official release package. The downfall of course is the github one isn't signed and doesn't have the necessary generated files for people using autotools or static makefiles for platforms that don't support autotools or cmake.

FYI curetntly in case of dist tar ball this is not the case ..
Fragment of my spec file:
```spec
Summary:        A library that performs asynchronous DNS operations
Name:           c-ares
Version:        1.22.1
Release:        2%{?dist}
License:        MIT (https://spdx.org/licenses/MIT.html)
URL:            https://c-ares.haxx.se/
VCS:            https://github.com/c-ares/c-ares/
Source:         https://c-ares.haxx.se/download/%{name}-%{version}.tar.gz
Patch:          %{VCS}/commit/4982f76a.patch#/%{name}-Query-Cache-support-625.patch
Patch:          %{VCS}/commit/320cefe1.patch#/%{name}-ares_set_servers_-should-allow-an-empty-server-list.patch
Patch:          %{VCS}/commit/3db69bbf.patch#/%{name}-Fix-Windows-UWP-Store-building-and-add-to-CI-CD-627.patch
Patch:          %{VCS}/commit/6b408a9b.patch#/%{name}-clang-format.patch
Patch:          %{VCS}/commit/5c1a5d80.patch#/%{name}-cleanup-some-Windows-warnings-in-test.patch
Patch:          %{VCS}/commit/ecf7a7f1.patch#/%{name}-fix-additional-windows-test-warnings.patch
Patch:          %{VCS}/commit/33ee6c05.patch#/%{name}-CMake-CXXFLAGS-environment-wasn-t-being-read-because.patch
Patch:          %{VCS}/commit/5262da7e.patch#/%{name}-now-that-warnings-are-enabled-on-test-cases-clear-a-.patch
Patch:          %{VCS}/commit/00f7cf6d.patch#/%{name}-fix-test-case-regression-due-to-missing-parens.patch
Patch:          %{VCS}/commit/edfb94d4.patch#/%{name}-SonarCloud-fix-minor-codesmells.patch
```
and test with initialize buuild
```console
[tkloczko@pers-jacek SPECS]$ rpmbuild -bp c-ares.spec --nodeps
warning: Downloading https://c-ares.haxx.se/download/c-ares-1.22.1.tar.gz to /home/tkloczko/rpmbuild/SOURCES/c-ares-1.22.1.tar.gz
warning: Downloading https://github.com/c-ares/c-ares//commit/4982f76a.patch#/c-ares-Query-Cache-support-625.patch to /home/tkloczko/rpmbuild/SOURCES/c-ares-Query-Cache-support-625.patch
warning: Downloading https://github.com/c-ares/c-ares//commit/320cefe1.patch#/c-ares-ares_set_servers_-should-allow-an-empty-server-list.patch to /home/tkloczko/rpmbuild/SOURCES/c-ares-ares_set_servers_-should-allow-an-empty-server-list.patch
warning: Downloading https://github.com/c-ares/c-ares//commit/3db69bbf.patch#/c-ares-Fix-Windows-UWP-Store-building-and-add-to-CI-CD-627.patch to /home/tkloczko/rpmbuild/SOURCES/c-ares-Fix-Windows-UWP-Store-building-and-add-to-CI-CD-627.patch
warning: Downloading https://github.com/c-ares/c-ares//commit/6b408a9b.patch#/c-ares-clang-format.patch to /home/tkloczko/rpmbuild/SOURCES/c-ares-clang-format.patch
warning: Downloading https://github.com/c-ares/c-ares//commit/5c1a5d80.patch#/c-ares-cleanup-some-Windows-warnings-in-test.patch to /home/tkloczko/rpmbuild/SOURCES/c-ares-cleanup-some-Windows-warnings-in-test.patch
warning: Downloading https://github.com/c-ares/c-ares//commit/ecf7a7f1.patch#/c-ares-fix-additional-windows-test-warnings.patch to /home/tkloczko/rpmbuild/SOURCES/c-ares-fix-additional-windows-test-warnings.patch
warning: Downloading https://github.com/c-ares/c-ares//commit/33ee6c05.patch#/c-ares-CMake-CXXFLAGS-environment-wasn-t-being-read-because.patch to /home/tkloczko/rpmbuild/SOURCES/c-ares-CMake-CXXFLAGS-environment-wasn-t-being-read-because.patch
warning: Downloading https://github.com/c-ares/c-ares//commit/5262da7e.patch#/c-ares-now-that-warnings-are-enabled-on-test-cases-clear-a-.patch to /home/tkloczko/rpmbuild/SOURCES/c-ares-now-that-warnings-are-enabled-on-test-cases-clear-a-.patch
warning: Downloading https://github.com/c-ares/c-ares//commit/00f7cf6d.patch#/c-ares-fix-test-case-regression-due-to-missing-parens.patch to /home/tkloczko/rpmbuild/SOURCES/c-ares-fix-test-case-regression-due-to-missing-parens.patch
warning: Downloading https://github.com/c-ares/c-ares//commit/edfb94d4.patch#/c-ares-SonarCloud-fix-minor-codesmells.patch to /home/tkloczko/rpmbuild/SOURCES/c-ares-SonarCloud-fix-minor-codesmells.patch
warning: source_date_epoch_from_changelog set but %changelog is missing
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.nAwRj0
+ umask 022
+ cd /home/tkloczko/rpmbuild/BUILD
+ cd /home/tkloczko/rpmbuild/BUILD
+ rm -rf c-ares-1.22.1
+ /usr/lib/rpm/rpmuncompress -x /home/tkloczko/rpmbuild/SOURCES/c-ares-1.22.1.tar.gz
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd c-ares-1.22.1
+ rm -rf /home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/SPECPARTS
+ /usr/bin/mkdir -p /home/tkloczko/rpmbuild/BUILD/c-ares-1.22.1/SPECPARTS
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ /usr/lib/rpm/rpmuncompress /home/tkloczko/rpmbuild/SOURCES/c-ares-Query-Cache-support-625.patch
+ /usr/bin/patch -p1 -s --fuzz=0 --no-backup-if-mismatch -f
+ /usr/lib/rpm/rpmuncompress /home/tkloczko/rpmbuild/SOURCES/c-ares-ares_set_servers_-should-allow-an-empty-server-list.patch
+ /usr/bin/patch -p1 -s --fuzz=0 --no-backup-if-mismatch -f
+ /usr/lib/rpm/rpmuncompress /home/tkloczko/rpmbuild/SOURCES/c-ares-Fix-Windows-UWP-Store-building-and-add-to-CI-CD-627.patch
+ /usr/bin/patch -p1 -s --fuzz=0 --no-backup-if-mismatch -f
The text leading up to this was:
--------------------------
|diff --git a/appveyor.yml b/appveyor.yml
|index b4572c8c7..45b4444b7 100644
|--- a/appveyor.yml
|+++ b/appveyor.yml
--------------------------
No file to patch.  Skipping patch.
3 out of 3 hunks ignored
error: Bad exit status from /var/tmp/rpm-tmp.nAwRj0 (%prep)
```

**Comment by bagder** [comment]
> FYI curetntly in case of dist tar ball this is not the case ..

Then please tell me which source files that are different between dist the tarball and the tagged source code dir. The `CMakeLists.txt` should be the only one.

**Comment by kloczek** [comment]
> don't think this can be securely automated really, hosting signing keys in github secrets for a github action to run just sounds like a compromise waiting to happen.

I;m not suggesting to switch completely to tar balls autograted tar balls.
I;m only asking to change tagging convention to make packaging easier.
Note about possibility of signing commits and tags it was only side note showing that many maintained abandoned completely providing dist tar balls because gitlab/github offers simpler possibilities.

Issue is that in case c-ares forcing to use only provided fist tar balls closes possibilities of quick integration some emergency patches downloaded directly from git repo commits.
Question only is: do you care about such details like how handy results of your work are or not?

And about security .. please read carefully documentation which ULR I've gave you about what offers signing tags.

**Comment by kloczek** [comment]
> > FYI cementy in case of dist tar ball this is not the case ..
> 
> Then please tell me which source files that are different between the dist tarball and the tagged source code dir. The `CMakeLists.txt` should be the only one.

Reject message is "No file to patch.  Skipping patch." and it points clearly on `appveyor.yml`.
You have two choices:
- remember about not mixing general commits with files modyfikatorów of some files not provided in dist tar ball (in such case would be possible skip such patches in patch sequence)
- provide as dist tar ball 100% files in tagged tar ball

And going to main topic of tagging convention:  I can understand that not changing such convention may be justified if already working automation uses current pattern. In such case it is will possible to ad two tags on the same has (some maintainers are using that approach),
If there is no such use case for current convention it should be technically no obstacles to change convention.

**Comment by bradh352** [comment]
Right, only the "code" is the same.  We don't distribute CI/CD files as those aren't relevant to the library itself.  That said, I could see an argument to ensure a single commit doesn't include both CI/CD and code changes so that patches are easier to pull in and apply.

**Comment by kloczek** [comment]
What you wrote does not change anything in context of the advices.

Just please remember that in many cases on top of the release is necessary to apply some later added commits patches.
We are living in the world in which we are dealing only with no-enough-tested-software and bugs/issues happens.
More important is in such case how quickly is possible to fix the issue using official or not official fix without waiting on maintainer to make next release.

**Comment by bradh352** [comment]
I still don't think it makes sense to place CI/CD tooling into a distribution tarball.

I could see a benefit in maintaining branches for c-ares which we currently don't do, specifically to backport bugfixes to the current stable release.  Likely CI/CD files wouldn't be touched in a backport and you could be sure you're only pulling in bugfixes and not features.

c-ares has been undergoing some pretty significant changes the last couple of months that does mean backporting only a bugfix is much harder for distributions.  

The downside of course is a distribution then asking for how long we are going to maintain any given branch, which we really don't have any resources for.

**Comment by kloczek** [comment]
> I still don't think it makes sense to place CI/CD tooling into a distribution tarball.
> 
> I could see a benefit in maintaining branches for c-ares which we currently don't do, specifically to backport bugfixes to the current stable release. Likely CI/CD files wouldn't be touched in a backport and you could be sure you're only pulling in bugfixes and not features.

Some CI specific build procedures should not be implemented in OOTB test suite or at least some CI very specific units/bits by default should be disabled.
Testing from many years is part of the build procedures used by package management software because executing test suites allow at least release packages without mos trivial issues usually catchable by OOTB test suites.

Test suite execution should be as well kind of step for whoever who is trying to submit any change to any project.

Really **please** have look on the subject from a bit different angle ..

**Comment by kloczek** [comment]
Just tested 1.26.0 and it is yet another issue.
Looks like `gmock` detection is missing
<detsils>

```console
[ 98%] Linking C executable ../../bin/adig
cd /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/x86_64-redhat-linux-gnu/src/tools && /usr/bin/cmake -E cmake_link_script CMakeFiles/adig.dir/link.txt --verbose=1
/usr/bin/gcc -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -O2 -g -grecord-gcc-switches -pipe -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fdata-sections -ffunction-sections -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -flto=auto -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -Wall -Werror=format-security -Wall -Wextra -Wcast-align -Wconversion -Wdeclaration-after-statement -Wdouble-promotion -Wfloat-equal -Wformat-security -Winit-self -Wjump-misses-init -Wlogical-op -Wmissing-braces -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-prototypes -Wnested-externs -Wno-coverage-mismatch -Wold-style-definition -Wpacked -Wpointer-arith -Wredundant-decls -Wshadow -Wsign-conversion -Wstrict-overflow -Wstrict-prototypes -Wtrampolines -Wundef -Wunused -Wvariadic-macros -Wvla -Wwrite-strings -Wimplicit-fallthrough=3 -Werror=implicit-int -Werror=implicit-function-declaration -DNDEBUG -fno-omit-frame-pointer -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -Wl,--gc-sections -Wl,--as-needed -Wl,--build-id=sha1 -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,-z,pack-relative-relocs -flto=auto -fuse-linker-plugin CMakeFiles/adig.dir/adig.c.o CMakeFiles/adig.dir/ares_getopt.c.o CMakeFiles/adig.dir/__/lib/ares_strcasecmp.c.o -o ../../bin/adig  -Wl,-rpath,/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/x86_64-redhat-linux-gnu/lib64: ../../lib64/libcares.so.2.11.0
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-naptr.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-uri.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:205: test/CMakeFiles/arestest.dir/ares-test-parse-naptr.cc.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:135: test/CMakeFiles/arestest.dir/ares-test-parse.cc.o] Error 1
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-a.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:303: test/CMakeFiles/arestest.dir/ares-test-parse-uri.cc.o] Error 1
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-ptr.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:149: test/CMakeFiles/arestest.dir/ares-test-parse-a.cc.o] Error 1
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-soa.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:233: test/CMakeFiles/arestest.dir/ares-test-parse-ptr.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:247: test/CMakeFiles/arestest.dir/ares-test-parse-soa.cc.o] Error 1
gmake[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/x86_64-redhat-linux-gnu'
gmake[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/x86_64-redhat-linux-gnu'
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-mock-ai.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-ai.h:31:10: fatal error: gmock/gmock.h: No such file or directory
   31 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
[ 98%] Built target aresfuzz
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:359: test/CMakeFiles/arestest.dir/ares-test-mock-ai.cc.o] Error 1
[ 98%] Built target aresfuzzname
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-mx.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.cc:29:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-main.cc:29:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-ns.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-aaaa.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-internal.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:191: test/CMakeFiles/arestest.dir/ares-test-parse-mx.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:107: test/CMakeFiles/arestest.dir/ares-test.cc.o] Error 1
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/dns-proto-test.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-mock-et.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-caa.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
compilation terminated.
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-init.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-soa-any.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:79: test/CMakeFiles/arestest.dir/ares-test-main.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:121: test/CMakeFiles/arestest.dir/ares-test-ns.cc.o] Error 1
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-live.cc:31:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-srv.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-txt.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:163: test/CMakeFiles/arestest.dir/ares-test-parse-aaaa.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:387: test/CMakeFiles/arestest.dir/ares-test-internal.cc.o] Error 1
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-misc.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-mock.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:261: test/CMakeFiles/arestest.dir/ares-test-parse-soa-any.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:289: test/CMakeFiles/arestest.dir/ares-test-parse-txt.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:93: test/CMakeFiles/arestest.dir/ares-test-init.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:177: test/CMakeFiles/arestest.dir/ares-test-parse-caa.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:373: test/CMakeFiles/arestest.dir/ares-test-mock-et.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:415: test/CMakeFiles/arestest.dir/dns-proto-test.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:331: test/CMakeFiles/arestest.dir/ares-test-live.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:275: test/CMakeFiles/arestest.dir/ares-test-parse-srv.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:317: test/CMakeFiles/arestest.dir/ares-test-misc.cc.o] Error 1
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:345: test/CMakeFiles/arestest.dir/ares-test-mock.cc.o] Error 1
In file included from /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test-parse-ns.cc:26:
/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/test/ares-test.h:40:10: fatal error: gmock/gmock.h: No such file or directory
   40 | #include "gmock/gmock.h"
      |          ^~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/x86_64-redhat-linux-gnu'
gmake[2]: *** [test/CMakeFiles/arestest.dir/build.make:219: test/CMakeFiles/arestest.dir/ares-test-parse-ns.cc.o] Error 1
[ 98%] Built target ahost
gmake[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/x86_64-redhat-linux-gnu'
[ 98%] Built target adig
gmake[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/x86_64-redhat-linux-gnu'
```
</detsils>

gmock has only gmock.pc pkgconfig file.
After add gmock devel to build env test suite is OK
```console
+ cd c-ares-1.26.0
+ /usr/bin/ctest --test-dir x86_64-redhat-linux-gnu --output-on-failure --force-new-ctest-process -j48
Internal ctest changing into directory: /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/x86_64-redhat-linux-gnu
Test project /home/tkloczko/rpmbuild/BUILD/c-ares-1.26.0/x86_64-redhat-linux-gnu
    Start 1: arestest
    Start 2: aresfuzz
    Start 3: aresfuzzname
1/3 Test #3: aresfuzzname .....................   Passed    0.00 sec
2/3 Test #2: aresfuzz .........................   Passed    0.08 sec
3/3 Test #1: arestest .........................   Passed  250.22 sec

100% tests passed, 0 tests failed out of 3
```

**Comment by bradh352** [comment]
The issue is really that gtest and gmock get split into separate packages on some systems which is strange, and the detection for gtest doesn't allow the components keyword.  I've added an explicit check in 906d2c1

**Comment by kloczek** [comment]
As I wrote switching to pkgconfig based detection will solve this issue.

**Comment by bradh352** [comment]
find_package() has builtin support for stuff like that.  Gmock is considered a component of GTest.  Did you actually try my commit to see if it resolved your complaint ?
